src/node_file.cc

```diff
@@ -1872,6 +1898,41 @@ static void Open(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+static void OpenFile(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  const int argc = args.Length();
+  CHECK_GE(argc, 4);
+
+  BufferValue path(env->isolate(), args[0]);
+  CHECK_NOT_NULL(*path);
+
+  CHECK(args[1]->IsInt32());
+  const int flags = args[1].As<Int32>()->Value();
+
+  CHECK(args[2]->IsInt32());
+  const int mode = args[2].As<Int32>()->Value();
+
+  CHECK(Buffer::HasInstance(args[3]));
+  Local<Object> buffer_obj = args[3].As<Object>();
+  char* fd_buffer = Buffer::Data(buffer_obj);
+
+  FSReqBase* req_wrap = GetReqWrap(args, 4, false);
+  if (req_wrap != nullptr) {
+    req_wrap->set_is_plain_open(true);
+    // Note: set fd in context
+    req_wrap->req()->data = fd_buffer;
+
+    FS_ASYNC_TRACE_BEGIN0(UV_FS_OPEN, req_wrap)
+    uv_fs_open(env->event_loop(),
```

Comment:
These should be using `AsyncCall`, otherwise be calling `req_wrap->Dispatch()`, not the libuv function directly (this is probably the cause of the crash you were seeing

Reply:
I already tried `AsyncCall,` but as he needs `args` as an argument, it was not convenient to use it in the `open` callback.

But I'll try it again and yeah, maybe using `req_wrap->Dispatch()` and `req_wrap->Init()` is probably a good alternative ^^

Reply:
I also remember that I've tried (in another [branch](https://github.com/nodejs/node/compare/main...tony-go:node:read-file-discovery-2#diff-6b7af086ab0b2132e8251bcb62b0f019fc95823f9ce32f949165af0d5f2cb7f2R288-R318)) to change `AsyncCall` to chain multiple fs calls. But as `AsyncCall` was widely used, I thought using something standalone could reduce the impact on other fs APIs. At least to run a benchmark and see if it is worth it.